core/server/worker/src/main/java/alluxio/worker/block/MonoBlockStore.java

```diff
@@ -264,8 +279,70 @@ public void requestSpace(long sessionId, long blockId, long additionalBytes)
     mLocalBlockStore.requestSpace(sessionId, blockId, additionalBytes);
   }
 
+  @Override
+  public List<BlockStatus> load(List<Block> blocks, String tag, OptionalInt bandwidth) {
+    ArrayList<CompletableFuture<BlockStatus>> futures = new ArrayList<>();
+    for (Block block : blocks) {
+      CompletableFuture<Void> future = CompletableFuture.runAsync(() -> {
+        try {
+          loadInternal(block.getBlockId(), block.getBlockSize(), block.getMountId(),
+              block.getUfsPath(), tag, bandwidth, block.getOffsetInFile());
+        } catch (Exception e) {
+          throw AlluxioRuntimeException.from(e);
+        }
+      }, GrpcExecutors.BLOCK_READER_EXECUTOR);
+      CompletableFuture<BlockStatus> exceptionally = future.handle((res, throwable) -> {
+        if (throwable != null) {
+          return BlockStatus.newBuilder().setBlock(block)
+              .setCode(AlluxioRuntimeException.from(throwable).getStatus().getCode().value())
+              .setMessage(throwable.getMessage()).setRetryable(false).build();
+        } else {
+          return null;
+        }
+      });
+      futures.add(exceptionally);
+    }
+    return futures.stream().map(CompletableFuture::join).filter(Objects::nonNull)
```

Comment:
Should we set successfully loaded blocks with some indication, like using Code.OK for code field inside BlockStatus?

Reply:
we only returns failures so we only deal with exceptional block

Reply:
Ah I see